Flaky test testNonDeltaTablesCannotBeAccessed & testHiveViewsCannotBeAccessed in TestDeltaLakeMinioAndHmsConnectorSmokeTest: "Schema 'test_schema....' does not exist"

https://github.com/trinodb/trino/actions/runs/6905645416/job/18789278949?pr=19797

```
Error:  Tests run: 127, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 145.8 s <<< FAILURE! -- in io.trino.plugin.deltalake.TestDeltaLakeMinioAndHmsConnectorSmokeTest
Error:  io.trino.plugin.deltalake.TestDeltaLakeMinioAndHmsConnectorSmokeTest.testNonDeltaTablesCannotBeAccessed -- Time elapsed: 13.90 s <<< ERROR!
io.trino.testing.QueryFailedException: line 1:1: Schema 'test_schemauf1ow1a40h' does not exist
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:133)
	at io.trino.testing.DistributedQueryRunner.executeWithQueryId(DistributedQueryRunner.java:502)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:495)
	at io.trino.testing.AbstractTestQueryFramework.computeActual(AbstractTestQueryFramework.java:313)
	at io.trino.testing.AbstractTestQueryFramework.computeScalar(AbstractTestQueryFramework.java:323)
	at io.trino.testing.AbstractTestQueryFramework.computeScalar(AbstractTestQueryFramework.java:318)
	at io.trino.plugin.deltalake.BaseDeltaLakeConnectorSmokeTest.testNonDeltaTablesCannotBeAccessed(BaseDeltaLakeConnectorSmokeTest.java:497)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.tryRemoveAndExec(ForkJoinPool.java:1351)
	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:422)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:651)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
	Suppressed: java.lang.Exception: SQL: SHOW TABLES FROM test_schemauf1ow1a40h LIKE 'hive_table'
		at io.trino.testing.DistributedQueryRunner.executeWithQueryId(DistributedQueryRunner.java:506)
		... 17 more
Caused by: io.trino.spi.TrinoException: line 1:1: Schema 'test_schemauf1ow1a40h' does not exist
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
	at io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowTables(ShowQueriesRewrite.java:290)
	at io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowTables(ShowQueriesRewrite.java:229)
	at io.trino.sql.tree.ShowTables.accept(ShowTables.java:72)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.rewrite.ShowQueriesRewrite.rewrite(ShowQueriesRewrite.java:226)
	at io.trino.sql.rewrite.StatementRewrite.rewrite(StatementRewrite.java:54)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:92)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:86)
	at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:274)
	at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:209)
	at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:850)
	at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:153)
	at io.trino.$gen.Trino_testversion____20231117_152301_6598.call(Unknown Source)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

https://github.com/trinodb/trino/actions/runs/6906231508/job/18790888931?pr=19800 

```
Error:  io.trino.plugin.deltalake.TestDeltaLakeMinioAndHmsConnectorSmokeTest.testNonDeltaTablesCannotBeAccessed -- Time elapsed: 14.70 s <<< ERROR!
io.trino.testing.QueryFailedException: line 1:1: Schema 'test_schema1xhz29z987' does not exist
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:133)
	at io.trino.testing.DistributedQueryRunner.executeWithQueryId(DistributedQueryRunner.java:502)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:495)
	at io.trino.testing.AbstractTestQueryFramework.computeActual(AbstractTestQueryFramework.java:313)
	at io.trino.testing.AbstractTestQueryFramework.computeScalar(AbstractTestQueryFramework.java:323)
	at io.trino.testing.AbstractTestQueryFramework.computeScalar(AbstractTestQueryFramework.java:318)
	at io.trino.plugin.deltalake.BaseDeltaLakeConnectorSmokeTest.testNonDeltaTablesCannotBeAccessed(BaseDeltaLakeConnectorSmokeTest.java:497)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.tryRemoveAndExec(ForkJoinPool.java:1351)
	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:422)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:651)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
	Suppressed: java.lang.Exception: SQL: SHOW TABLES FROM test_schema1xhz29z987 LIKE 'hive_table'
		at io.trino.testing.DistributedQueryRunner.executeWithQueryId(DistributedQueryRunner.java:506)
		... 17 more
Caused by: io.trino.spi.TrinoException: line 1:1: Schema 'test_schema1xhz29z987' does not exist
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
	at io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowTables(ShowQueriesRewrite.java:290)
	at io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowTables(ShowQueriesRewrite.java:229)
	at io.trino.sql.tree.ShowTables.accept(ShowTables.java:72)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.rewrite.ShowQueriesRewrite.rewrite(ShowQueriesRewrite.java:226)
	at io.trino.sql.rewrite.StatementRewrite.rewrite(StatementRewrite.java:54)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:92)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:86)
	at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:274)
	at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:209)
	at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:850)
	at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:153)
	at io.trino.$gen.Trino_testversion____20231117_160947_6515.call(Unknown Source)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

--- a/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java
+++ b/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java
@@ -488,18 +488,6 @@ public abstract class BaseDeltaLakeConnectorSmokeTest
     }
 
     @Test
-    public void testNonDeltaTablesCannotBeAccessed()
-    {
-        String schemaName = "test_schema" + randomNameSuffix();
-        String tableName = "hive_table";
-        hiveHadoop.runOnHive("CREATE DATABASE " + schemaName);
-        hiveHadoop.runOnHive(format("CREATE TABLE %s.%s (id BIGINT)", schemaName, tableName));
-        assertThat(computeScalar(format("SHOW TABLES FROM %s LIKE '%s'", schemaName, tableName))).isEqualTo(tableName);
-        assertThatThrownBy(() -> computeActual("DESCRIBE " + schemaName + "." + tableName)).hasMessageContaining(tableName + " is not a Delta Lake table");
-        hiveHadoop.runOnHive("DROP DATABASE " + schemaName + " CASCADE");
-    }
-
-    @Test
     public void testDropDatabricksTable()
     {
         testDropTable(

--- a/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeSharedMetastoreViewsTest.java
+++ b/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeSharedMetastoreViewsTest.java
@@ -35,6 +35,7 @@ import static com.google.inject.util.Modules.EMPTY_MODULE;
 import static io.trino.testing.TestingNames.randomNameSuffix;
 import static io.trino.testing.TestingSession.testSessionBuilder;
 import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
 
 /**
@@ -154,6 +155,23 @@ public abstract class BaseDeltaLakeSharedMetastoreViewsTest
         finally {
             assertUpdate(format("DROP TABLE IF EXISTS %s", hiveTable));
             assertUpdate(format("DROP VIEW IF EXISTS %s", trinoViewOnHive));
+        }
+    }
+
+    @Test
+    public void testNonDeltaTablesCannotBeAccessed()
+    {
+        String schemaName = "test_schema" + randomNameSuffix();
+        String tableName = "hive_table";
+
+        assertUpdate("CREATE SCHEMA %s.%s".formatted(HIVE_CATALOG_NAME, schemaName));
+        try {
+            assertUpdate("CREATE TABLE %s.%s.%s(id BIGINT)".formatted(HIVE_CATALOG_NAME, schemaName, tableName));
+            assertThat(computeScalar(format("SHOW TABLES FROM %s LIKE '%s'", schemaName, tableName))).isEqualTo(tableName);
+            assertQueryFails("DESCRIBE " + schemaName + "." + tableName, ".* is not a Delta Lake table");
+        }
+        finally {
+            assertUpdate("DROP SCHEMA %s.%s CASCADE".formatted(HIVE_CATALOG_NAME, schemaName));
         }
     }
 

